lib/msal-common/src/unifiedCache/entities/Credential.ts

```diff
@@ -3,13 +3,62 @@
  * Licensed under the MIT License.
  */
 
+import { Separators, CredentialType } from "../../utils/Constants";
+
 /**
  * Base type for credentials to be stored in the cache: eg: ACCESS_TOKEN, ID_TOKEN etc
  */
 export class Credential {
     homeAccountId: string;
     environment: string;
-    credentialType: string;
+    credentialType: CredentialType;
     clientId: string;
     secret: string;
-};
+    familyId?: string;
+    realm?: string;
+    target?: string;
+
+    /**
+     * Generate Account Id key component as per the schema: <home_account_id>-<environment>
+     */
+    generateAccountId(): string {
```

Comment:
Can these be used as utils for cache lookup (i.e. generate keys for looking up a specific cred)?

Reply:
So generating accountId for an account object instance should be separate from generating it from a set of values. I will try coming up with common code as needed, for now I added static helpers in `CacheHelper.ts` in a later PR to address this. Point noted though.